src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java

```diff
@@ -118,6 +120,30 @@ public int getAmountOfLayers() {
     return nLayers;
   }
 
+  /**
+   * Returns the selected layer. Returns {@code null} if there is no selected layer.
+   *
+   * @return the selected layer.
+   */
+  public IReadOnlyLayer getSelectedLayer() {
+    return selectedLayer;
+  }
+
+  /**
+   * Sets the the selected layer by its index. Does nothing if the index is outside the range: [0,
+   * amount of layers).
+   *
+   * @param index the index of the selected layer.
+   */
+  void setSelectedLayer(int index) {
+    int i = 0;
+    for (IReadOnlyLayer layer : layers) {
+      if (i++ == index) {
```

Comment:
Might make this more readable :neckbeard:

Reply:
This.